### ISSUE TITLE
Support libclang

### ISSUE THREAD

**Issue by LukePrior** [pyodide/pyodide#2357]
## 🐍 Package Request

- Package Name and Version : libclang 13.0.0

- Package URL : https://pypi.org/project/libclang/

**Comment by hoodmane** [comment]
I think this is pretty far out of scope. Building this is probably harder than everything else we do combined...

**Comment by LukePrior** [comment]
Ok no worries I will go for a different approach in that case.

One last thing however, I can install clang using micropip:

```
import micropip
await micropip.install('clang')
import clang.cindex
```

But obviously, this doesn't work as clang is not installed so when running

```
index = clang.cindex.Index.create()
```

I get the following error:

```
OSError: Could not load dynamic lib: libclang-14.so
Error: FS error. To provide a path to libclang use Config.set_library_path() or Config.set_library_file().
```

I'm guessing that I would need a version of clang compiled to WASM which I could then import using requests per the FAQ.

**Comment by hoodmane** [comment]
Yeah hypothetically that is what you would need to do. I don't know if you've ever built LLVM before, but it's rather involved. We've discussed before the possibility of building clang from source to solve some of our platform-related difficulties, but we decided that building native x86_64-linux clang was outside of our maintenance budget for Pyodide.

**Comment by TheComputerM** [comment]
This might be of help, https://github.com/TheComputerM/libclang-wasm as it compiles libclang to WASM and exposes its functions.
